pkg/controller/atlascluster/atlascluster_controller.go

```diff
@@ -114,28 +112,41 @@ func (r *AtlasClusterReconciler) Reconcile(context context.Context, req ctrl.Req
 	}
 	ctx.Client = atlasClient
 
+	handleCluster := r.selectClusterHandler(cluster)
+	if result, _ := handleCluster(ctx, project, cluster); !result.IsOk() {
+		ctx.SetConditionFromResult(status.ClusterReadyType, result)
+		return result.ReconcileResult(), nil
+	}
+
+	if result := r.handleAdvancedOptions(ctx, project, cluster); !result.IsOk() {
+		ctx.SetConditionFromResult(status.ClusterReadyType, result)
+		return result.ReconcileResult(), nil
+	}
```

Comment:
do we always want to handle advanced options? How does this behave when we are dealing with non-advanced clusters?

Reply:
Advanced options are not related to the Advanced Clusters and can be set/changed for both cluster types.